tests/gun_aiming.cpp

```diff
@@ -8,19 +8,16 @@ static void test_internal( const npc& who, const item &gun )
 {
     THEN( "the effective range is correctly calcuated" ) {
         // calculate range for 50% chance of critical hit at arbitrary recoil
-        double recoil = rng_float( 0, 1000 );
-        double range = who.gun_current_range( gun, recoil, 50, accuracy_critical );
+        for ( double recoil = 0; recoil < 1000; recoil += 100 ) {
+            double range = who.gun_current_range( gun, recoil, 50, accuracy_critical );
```

Comment:
Yes I looked at that solution first but you can only pass base recoil. A solution with nested loops checking various values of `recoil` and `chance` would be best